typings/index.d.ts

```diff
@@ -0,0 +1,1909 @@
+
+  // From `/common`
+  export const black: string;
+  export const white: string;
+
+  export const darkBlack: string;
+  export const darkWhite: string;
+  export const faintBlack: string;
+  export const fullBlack: string;
+  export const fullWhite: string;
+  export const lightBlack: string;
+  export const lightWhite: string;
+  export const minBlack: string;
+  export const transparent: string;
+}
+
+/* ============================================= */
+/*                                               */
+/*                   INTERNAL                    */
+/*                                               */
+/* ============================================= */
+
+declare module 'material-ui/internal' {
+  /**
+   * NOTE: There is much more inside this module,
+   * but not sure if this should be exposed or not.
+   */
+}
+
+declare module 'material-ui/internal/Transition' {
+  export type TransitionCallback = (element: HTMLElement) => void;
+  export type TransitionRequestTimeout = (element: HTMLElement) => number;
+
+  export type TransitionHandlers = {
+    onEnter: TransitionCallback;
+    onEntering: TransitionCallback;
+    onEntered: TransitionCallback;
+    onExit: TransitionCallback;
+    onExiting: TransitionCallback;
+    onExited: TransitionCallback;
+  };
+
+  export interface TransitionProps extends Partial<TransitionHandlers> {
+    children?: React.ReactElement<any>;
+    className?: string;
+    enteredClassName?: string;
+    enteringClassName?: string;
+    exitedClassName?: string;
+    exitingClassName?: string;
+    in?: boolean;
+    onRequestTimeout?: TransitionRequestTimeout;
+    timeout?: number;
+    transitionAppear?: boolean;
+    unmountOnExit?: boolean;
+  }
+
+  export default class Transition extends React.Component<TransitionProps> {}
+}
+
+/* ============================================= */
+/*                                               */
+/*                     STYLES                    */
+/*                                               */
+/* ============================================= */
+
+declare module 'material-ui/styles' {
+  export {
+    default as MuiThemeProvider,
+  } from 'material-ui/styles/MuiThemeProvider';
+  export { default as createBreakpoints } from 'material-ui/styles/breakpoints';
+  export { default as createMuiTheme } from 'material-ui/styles/theme';
+  export { default as createPalette } from 'material-ui/styles/palette';
+  export {
+    default as createStyleSheet,
+  } from 'material-ui/styles/createStyleSheet';
+  export { default as createTypography } from 'material-ui/styles/typography';
+  export { default as withStyles } from 'material-ui/styles/withStyles';
+  export { default as withTheme } from 'material-ui/styles/withTheme';
+}
+
+declare module 'material-ui/styles/MuiThemeProvider' {
+  import { Theme } from 'material-ui/styles/theme';
+
+  export interface MuiThemeProviderProps {
+    theme?: Theme<any>;
+    sheetsManager?: Object;
```

Comment:
That's what the `PropTypes` say :)
https://github.com/callemall/material-ui/blob/73a6d33072eabd93d1c78e9ae1be781d21601c7c/src/styles/MuiThemeProvider.js#L77-L92